### DISCUSSION TITLE
aptos 的资源账户是做什么用的。

### DISCUSSION THREAD

**Discussion by v1xingyue** [NonceGeek/Move-Camp#224] [Q&A] [answered]
参考的这篇文章 : https://mp.weixin.qq.com/s?__biz=MzI0NTM0MzE5Mw==&mid=2247486957&idx=1&sn=51861dfdb61b20aaaf67a508399c7a77&chksm=e94ebb5bde39324d8909761437e1cd040fde7bb9c0887142f759c52ab5f2683e44990f5e4f6c&scene=21#wechat_redirect 

1.6创建资源账户。

为什么要创建资源账户呢？
资源账户有什么作用。

貌似没有创建资源账户，合约依然可以部署，执行。

**Comment by Demian101** [comment, marked as answer]
### Resource Accounts 资源账户

https://aptos.dev/guides/resource-accounts/

A resource account is used to manage resources independent of an account managed by a user.
资源帐户用于管理独立于用户管理的帐户的资源。

> Hector: "对于发行资源比如 NFT 和代币的情况下，将合约直接发布在用户地址上是极其不安全的，谁知道项目方会不会通过普通用户地址进行修改和操作某些东西。**资源账户类似于专有的管理账户，只能初始化一次，后续无法改动**"


In Aptos a resource account is created based upon the sha3 256 of the source's address and additional seed data.A resource account can only be created once, this is designated by setting the`Account::signer_capability_offer::for` to the address of the resource account. 
在 Aptos 中，基于源地址的 sha3 256 和附加 seed 数据创建资源帐户。资源帐户只能创建一次，这通过将`Account::signer_capability_offer::for` 设置为资源帐户。

While an entity may call`create_account` to attempt to claim an account ahead of the creation of a resource account, if found Aptos will transition ownership of the account over to the resource account. This is done by validating that the account hasyet to execute any transactions and that the `Account::signer_capability_offer::for` is none. The probability of acollision where someone has legitimately produced a private key that maps to a resource account address is lessthan `(1/2)^(256)`.
虽然实体可能会调用 `create_account` 来尝试在创建资源帐户之前 claim an accoun ，如果找到，Aptos 会将帐户的所有权转移到资源帐户。这是通过验证帐户尚未执行任何交易并且 `Account::signer_capability_offer::for` 为 `none` 来完成的。有人合法生成映射到资源帐户地址的私钥的冲突概率小于  `(1/2)^(256)`.

**Comment by Demian101** [comment]
@FF

这个资源账户（resource account）的功能，和普通账户没什么区别，一摸一样。
不过他没有公私钥。并且拥有signCap的概念。（谁拥有这个Cap，谁才能控制这个账户），一般这个用在专门的数据存储（比如NFT Market，每个店铺都拥有对应的resource account）以及权限鉴别分配（区分vip和普通用户）。
